dbms/src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -179,6 +180,27 @@ class MergeTreeData : public IStorage
     using DataPartsLock = std::unique_lock<std::mutex>;
     DataPartsLock lockParts() const { return DataPartsLock(data_parts_mutex); }
 
+    MergeTreeDataPartType choosePartType(size_t bytes_on_disk, size_t rows_count) const;
+
+    /// After this methods setColumns must be called
+    MutableDataPartPtr createPart(const String & name,
+        const MergeTreePartInfo & part_info, const DiskPtr & disk,
+        const NamesAndTypesList & columns,
+        size_t bytes_on_disk, size_t rows_num,
+        const String & relative_path) const;
+
+    MutableDataPartPtr createPart(const String & name,
+        MergeTreeDataPartType type, const MergeTreePartInfo & part_info,
+        const DiskPtr & disk, const String & relative_path) const;
+
+    /// After this methods 'loadColumnsChecksumsIndexes' must be called
+    /// FIXME make this inside this function
```

Comment:
I decided to left it as is, because createPart and loadColumnsChecksumsIndexes are such independent functions and it's better not complicate createPart definition with parameters, which just will be passed to other function.